src/core/RenderContext.cpp

```diff
@@ -130,17 +140,21 @@ void RenderContext::updateGLWindows()
 
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
+        window->setBlockDrawCalls( false );
         window->viewport()->repaint();
+        window->setBlockDrawCalls( true );
         ++activeGLWindowIndex_;
     }
+    glFinish();
 }
 
 void RenderContext::swapBuffers()
 {
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
-        static_cast<QGLWidget*>( window->viewport( ))->makeCurrent();
-        static_cast<QGLWidget*>( window->viewport( ))->swapBuffers();
+        QGLWidget* glContext = static_cast<QGLWidget*>( window->viewport( ));
+        glContext->makeCurrent();
+        glContext->swapBuffers();
```

Comment:
glFinish blocks execution. glFlush just ensures that the commands are executed right away without waiting for completion (a bit like TCP_NODELAY).